src/math/integerhistogram1d.h

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#pragma once
+
+#include "math/data1d.h"
+#include "math/sampleddouble.h"
+#include <map>
+
+// Pure Integer Histogram
+class IntegerHistogram1D
+{
+    public:
+    IntegerHistogram1D();
+    ~IntegerHistogram1D() = default;
+    IntegerHistogram1D(const IntegerHistogram1D &source);
+    // Clear data
+    void clear();
+
+    /*
+     * Histogram Data
+     */
+    private:
+    // Minimum value for data (hard left-edge of first bin)
+    int minimum_;
+    // Maximum value for data (hard right-edge of last bin, adjusted to match bin width if necessary)
+    int maximum_;
+    // Bin width
+    int binWidth_;
+    // Number of bins
+    int nBins_;
+    // Histogram bins
+    std::vector<long int> bins_;
+    // Array of bin centres
+    std::vector<int> binCentres_;
+    // Accumulated averages
+    std::vector<SampledDouble> averages_;
+    // Number of values binned over all bins
+    long int nBinned_;
+    // Number of points missed (out of bin range)
+    long int nMissed_;
+    // Accumulated data
+    Data1D accumulatedData_;
+    // key is the bin and value is the count
+    std::map<int, long int> histMap_;
```

Comment:
As mentioned above, we need to tie together the "instantaneous" binned values and their running averages, which was previously managed by two `std::vector`s (`bins_` and `averages_`) of the same size. My suggestion would be to use a `std::pair`.  Also renamed it to be `bins_` since we can use it with the old one removed.
```suggestion
    // Map of histogram bin values to instantaneous counts and accumulated averages
    std::map<int, std::pair<long int, SampledDouble>> bins_;
```